### PR TITLE
implement design for wallet page header when a wallet has been selected

### PR DESCRIPTION
**Pull request by dreacot**
closes https://github.com/crypto-power/cryptopower/issues/296

figma
Overview page (market overview   recent proposal off) (11)

this PR
Screenshot from 2023-12-06 20-53-58

### CODE REVIEW

File: ui/cryptomaterial/segmented_control.go [hunk 1]
```diff
@@ -75,7 +77,15 @@ func (sc *SegmentedControl) SetEnableSwipe(enable bool) {
...
-func (sc *SegmentedControl) Layout(gtx C, body func(gtx C) D) D {
+// Layout handles the segmented control's layout, it receives an optional isMobileView bool
+// parameter which is used to determine if the segmented control should be displayed in mobile view
+// or not. If the parameter is not provided, isMobileView defaults to false.
+func (sc *SegmentedControl) Layout(gtx C, body func(gtx C) D, isMobileView ...bool) D {
+	sc.isMobileView = false
+	if len(isMobileView) > 0 {
+		sc.isMobileView = isMobileView[0]
+	}
+
```
Comment (ukane-philemon) [inline ui/cryptomaterial/segmented_control.go:88]: Errm, can we consider `sc.isMobileView = len(isMobileView) > 0  && isMobileView[0]`?

File: ui/page/root/main_page.go [hunk 2]
```diff
@@ -456,73 +464,110 @@ func (swmp *SingleWalletMasterPage) LayoutTopBar(gtx C) D {
...
+										}.Layout(gtx,
+											layout.Rigid(func(gtx C) D {
+												image := components.CoinImageBySymbol(swmp.Load, assetType, isWatchOnlyWallet)
+												if image != nil {
+													if swmp.Load.IsMobileView() {
+														return image.Layout16dp(gtx)
+													}
+													return image.Layout24dp(gtx)
+												}
+												return D{}
```
Comment (ukane-philemon) [inline ui/page/root/main_page.go:509]:
```
if image == nil {
  return D{}
}

....


```

File: ui/page/root/main_page.go [hunk 3]
```diff
@@ -456,73 +464,110 @@ func (swmp *SingleWalletMasterPage) LayoutTopBar(gtx C) D {
...
+														Left: values.MarginPadding10,
+													}.Layout(gtx, func(gtx C) D {
+														textSize := values.TextSize16
+														if swmp.Load.IsMobileView() {
+															textSize = values.TextSize12
+														}
+														return components.WalletHightlighLabel(swmp.Theme, gtx, textSize, values.String(values.StrWatchOnly))
+													})
+												}
+												return D{}
```
Comment (ukane-philemon) [inline ui/page/root/main_page.go:533]: Same as my comment above.

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Good work @dreacot. A few nits:
I kinda feel the header text is not true centre. 

Screenshot 2023-12-07 at 4 37 51 PM

File: ui/page/root/home_page.go [hunk 4]
```diff
@@ -407,6 +409,13 @@ func (hp *HomePage) LayoutTopBar(gtx C) D {
 	padding20 := values.MarginPadding20
 	padding10 := values.MarginPadding10
 
+	topBottomPadding := padding10
+	// Remove top and bottom padding if on mobile view and on the SingleWalletMasterPage.
+	// this hides the gap between the top bar and the page content.
+	if hp.CurrentPageID() == MainPageID {
```
Comment (JustinBeBoy) [inline ui/page/root/home_page.go:415]: how do you know current app is mobile view ?

Comment (JustinBeBoy) [reply ui/page/root/home_page.go:415]: I don't see any check mobile view for it

File: ui/page/root/home_page.go [hunk 5]
```diff
@@ -415,11 +424,19 @@ func (hp *HomePage) LayoutTopBar(gtx C) D {
...
-			Bottom: padding10,
+			Top:    topBottomPadding,
+			Bottom: topBottomPadding,
 		},
 	}.Layout(gtx,
-		layout.Rigid(hp.totalBalanceLayout),
+		layout.Rigid(func(gtx C) D {
+			// Hide the total asset balance usd amount while on mobile view
+			// and on the SingleWalletMasterPage.
+			if hp.CurrentPageID() == MainPageID {
```
Comment (JustinBeBoy) [inline ui/page/root/home_page.go:434]: how do you know current app is mobile view ? I don't see any check mobile view for it

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
1. Desktop mode header has disappeared

image

**Comment by JustinBeBoy** [comment]
may be we need add scroll to segmented control, I have selected a page where the segmented header is not displayed

image

**Comment by dreacot** [comment]
> may be we need add scroll to segmented control, I have selected a page where the segmented header is not displayed
> 
> 
> 
> image
> 
> 

Yes, please create an issue for it

File: ui/page/root/home_page.go (hunk 4 above)
Comment (dreacot) [reply ui/page/root/home_page.go:415]: my bad the mobile view comment shouldn't be there

File: ui/page/root/home_page.go (hunk 5 above)
Comment (dreacot) [reply ui/page/root/home_page.go:434]: the mobile view comment shouldn't be there, thanks

**Comment by dreacot** [comment]
> Good work @dreacot. A few nits: I kinda feel the header text is not true centre.
> 
> Screenshot 2023-12-07 at 4 37 51 PM

it's centered after the back arrow, quite tricky to center it without the back arrow being factored in

**Comment by dreacot** [comment]
> > 1. Desktop mode header has disappeared
> > 
> > image
> 
> i cannot reproduce that
> 
> Screenshot from 2023-12-07 12-44-45

@ukane-philemon can you reproduce this?

**Comment by ukane-philemon** [comment]
> @ukane-philemon can you reproduce this?

Yh, I saw it one time and wanted to mention but somehow it didn’t happen again. I can't reproduce.

**Comment by JustinBeBoy** [comment]
> > @ukane-philemon can you reproduce this?
> 
> Yh, I saw it one time and wanted to mention but somehow it didn’t happen again. I can't reproduce.

If it happens once then it is likely to happen, we should try to find it

**Review by ukane-philemon — APPROVED** [review]
Looks good.

**Comment by dreacot** [comment]
> > > @ukane-philemon can you reproduce this?
> > 
> > 
> > Yh, I saw it one time and wanted to mention but somehow it didn’t happen again. I can't reproduce.
> 
> If it happens once then it is likely to happen, we should try to find it

yeah, but we need clear reproduction steps.

the only possibility i can think of is when if `hp.CurrentPageID() == MainPageID {` is true when on the overview page which should be impossible

we should keep an eye on it

**Review by ukane-philemon — APPROVED** [review]

**Review by JustinBeBoy — APPROVED** [review]

**Pull request merged**